Description: Fixing spec that fails sporadically on libxml 2.7.8 and later.
From: Mike Dalessio <mike@csa.net>
Bug: https://github.com/tenderlove/nokogiri/issues/626
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=661690
Origin: https://github.com/tenderlove/nokogiri/commit/1ea45940a1ebd78749485d8928d89720db586b3b
---
 test/xml/test_document.rb |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

--- a/test/xml/test_document.rb
+++ b/test/xml/test_document.rb
@@ -358,13 +358,17 @@
       def test_subclass_parse
         klass = Class.new(Nokogiri::XML::Document)
         doc = klass.parse(File.read(XML_FILE))
-        assert_equal @xml.to_s, doc.to_s
+        # lame hack uses root to avoid comparing DOCTYPE tags which can appear out of order.
+        # I should really finish lorax and use that here.
+        assert_equal @xml.root.to_s, doc.root.to_s
         assert_instance_of klass, doc
       end
 
       def test_document_parse_method
         xml = Nokogiri::XML::Document.parse(File.read(XML_FILE))
-        assert_equal @xml.to_s, xml.to_s
+        # lame hack uses root to avoid comparing DOCTYPE tags which can appear out of order.
+        # I should really finish lorax and use that here.
+        assert_equal @xml.root.to_s, xml.root.to_s
       end
 
       def test_encoding=
